### PR TITLE
Use CIS hardened images

### PR DESCRIPTION
**Pull request by nickumia-reisys**
Related to https://github.com/GSA/data.gov/issues/3668

### CODE REVIEW

File: terraform/modules/provision-aws/eks.tf [hunk 1]
```diff
@@ -110,6 +119,22 @@ module "eks" {
         }
       }
 
+      custom_ami = {
+        ami_id = aws_ami.gsa-ise.id
+        enable_bootstrap_user_data = true
+        bootstrap_extra_args = "--container-runtime containerd --kubelet-extra-args '--max-pods=20'"
```
Comment (mogul) [inline terraform/modules/provision-aws/eks.tf:125]: I see you planning ahead for the transition to containerd, but given that [it's not yet possible to set the default upstream repo for containerd](https://github.com/GSA/data.gov/issues/3747#issuecomment-1079980390), and it is for dockerd, maybe staying on dockerd is warranted for a while yet.

**Comment by nickumia-reisys** [comment]
I suspect that the test failed because of this, 
image

It was trying to start the ebs-app and just hung there, so it would make sense..

**Comment by nickumia-reisys** [comment]
This is the best screenshot that I have for proof that the issue was with there not being enough nodes to run 2048 and ebs test workloads on,
image

**Comment by mogul** [comment]
Could that have something to do with setting `'--max-pods=20'`? That would mean only 20 pods max would be scheduled on the single instance, even if it had capacity to schedule more.

**Comment by mogul** [comment]
(I don't know what the default is when you don't specify that argument.)

**Comment by nickumia-reisys** [comment]
That could definitely have an effect!  @mogul I could leave that option off.  I guess I hadn't thought about that haha..

**Comment by nickumia-reisys** [comment]
> (I don't know what the default is when you don't specify that argument.)

That's also true.  I mean, we could specify a really high number and then let the CPU and Memory determine if a pod can run.

**Comment by mogul** [comment]
It looks like the default varies by instance type, and they're [listed here](https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt).

**Review by mogul — APPROVED** [review]

**Pull request merged**